src/sage/rings/polynomial/laurent_polynomial_ideal.py

```diff
@@ -196,9 +198,17 @@ def __contains__(self, f):
         if not f or f in self.gens():
             return True
         f = self.ring()(f)
-        g = f.__reduce__()[1][0]
+        if self.ring().ngens() > 1:
+            g = f.__reduce__()[1][0]
+        else:
+            g = f.__reduce__()[1][1]
         return (g in self.polynomial_ideal())
 
+    def gens_reduced(self):
```

Comment:
This method needs documentation and tests.

Reply:
OK